hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java

```diff
@@ -252,6 +252,11 @@ private static void setEnvironment(String key, String value) throws Exception {
                 map.put(key, value);
             }
         }
+
+        Field field = Class.forName("java.lang.ProcessEnvironment").getDeclaredField("theCaseInsensitiveEnvironment");
```

Comment:
Like the part above it, it adds to faking the set environment variables.
Some methods use this case insensitive environment, some the other that is manipulated.
Maybe this is different between Windows and *nix Java versions, though I thought I tested on both.
So maybe I should add some failsafety like in the block above it.